src/NHibernate/Loader/Criteria/CriteriaLoader.cs

```diff
@@ -46,6 +50,10 @@ public partial class CriteriaLoader : OuterJoinLoader
 
 			InitFromWalker(walker);
 
+			if (translator.UncacheableCriteriaCollectionPersisters.Any())
```

Comment:
I don't quite agree with this check because this obfuscates things by making stuff dependant on a index. Although it looks as an optimization it produces a bigger array.

Reply:
Actually I didn't mean it as optimization but merely to make it processed the same way as others properties (see  `CollectionAliases`, 'CollectionOwners', `CollectionPersisters`,...).  And that's collection fetches -  it seems big arrays should not be an issue.